pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,531 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"cosmossdk.io/x/feegrant"
+	"github.com/celestiaorg/celestia-app/v6/pkg/appconsts"
+	"github.com/celestiaorg/go-square/v3/share"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// SubmissionJob represents a transaction submission task for parallel processing
+type SubmissionJob struct {
+	ID       string
+	Blobs    []*share.Blob
+	Options  []TxOption
+	Ctx      context.Context
+	ResultsC chan SubmissionResult
+}
+
+// SubmissionResult contains the result of a parallel transaction submission
+type SubmissionResult struct {
+	Signer     string
+	TxResponse *TxResponse
+	Error      error
+}
+
+// ParallelTxPool manages parallel transaction submission
+type ParallelTxPool struct {
+	mtx         sync.RWMutex
+	client      *TxClient
+	jobQueue    chan *SubmissionJob
```

Comment:
random comment, but the cool thing about using a queue and single worker is that if users need to get txs included in order, that is as close as get

still needs a flup if we want to provide guarantees around it, but generally should work well since all workers wait to submit a new tx until after they get a confirmation